presto-main/src/main/java/com/facebook/presto/operator/MultiChannelGroupByHash.java

```diff
@@ -204,40 +204,33 @@ public void appendValuesTo(int groupId, PageBuilder pageBuilder, int outputChann
     public void addPage(Page page)
     {
         if (canProcessDictionary(page)) {
-            addDictionaryPage(page);
+            boolean done = new AddDictionaryWork(page).process();
```

Comment:
cmt msg: "Adapt the new Work interface to MultiChannelGroupByHash."

Not sure "adapt" is the right word, since you're not changing the `Work` interface. "Apply"? "Use .. in"?